apps/system/js/core.js

```diff
@@ -58,6 +98,46 @@
           this.debug('API: ' + api + ' not found, skpping the handler.');
         }
       }
+
+      var self = this;
+      var idleObserver = {
+        time: 10,
+        onidle: function() {
+          navigator.removeIdleObserver(idleObserver);
+          self._startSideModules();
+          // XXX: find a proper launcher for these modules
+          LazyLoader.load([
+            'js/download/download_manager.js',
+            'js/payment.js',
+            'js/identity.js',
+            'js/devtools/logshake.js',
+            'js/devtools/remote_debugger.js',
+            'shared/js/date_time_helper.js'
+          ]).then(function() {
+            self.remoteDebugger = new RemoteDebugger();
+            // self.remoteDebugger.start();
+          });
+        }
+      };
+      navigator.addIdleObserver(idleObserver);
```

Comment:
Right now it looks like we have 3 places that use this 10s idle timer to do things like loading imports and starting side modules. Perhaps we should abstract this into BaseModule such that you could have things like:
```
Core.LOAD_ON_IDLE = [
  'js/download/download_manager.js',
  'js/payment.js',
  ...
];
Core.START_MODULES_ON_IDLE = [
  'NotificationScreen',
  'AirplaneMode',
  ...
];
```
and even perhaps have a function`__on_idle` that gets run after the IDLE modules have been loaded/started. In any case, it seems a little unfortunate that these modules have to repeat the idle observer logic, and yet all do very similar things.